cuqi/experimental/mcmc/_rto.py

```diff
@@ -0,0 +1,265 @@
+
+        # Check Linear model and Gaussian likelihood(s)
+        if isinstance(self.target, cuqi.distribution.Posterior):
+            if not isinstance(self.model, cuqi.model.LinearModel):
+                raise TypeError("Model needs to be linear")
+
+            if not hasattr(self.likelihood.distribution, "sqrtprec"):
+                raise TypeError("Distribution in Likelihood must contain a sqrtprec attribute")
+            
+        elif isinstance(self.target, cuqi.distribution.MultipleLikelihoodPosterior): # Elif used for further alternatives, e.g., stacked posterior
+            for likelihood in self.likelihoods:
+                if not isinstance(likelihood.model, cuqi.model.LinearModel):
+                    raise TypeError("Model needs to be linear")
+
+                if not hasattr(likelihood.distribution, "sqrtprec"):
+                    raise TypeError("Distribution in Likelihood must contain a sqrtprec attribute")
+        
+        # Check Gaussian prior
+        if not hasattr(self.prior, "sqrtprec"):
+            raise TypeError("prior must contain a sqrtprec attribute")
+
+        if not hasattr(self.prior, "sqrtprecTimesMean"):
+            raise TypeError("Prior must contain a sqrtprecTimesMean attribute")
+
+    def get_state(self): #TODO: LinearRTO only need initial_point for reproducibility?
+        return {'sampler_type': 'LinearRTO'}
+
+    def set_state(self, state): #TODO: LinearRTO only need initial_point for reproducibility?
+        pass
+
+class RegularizedLinearRTONew(LinearRTONew):
+    """
+    Regularized Linear RTO (Randomize-Then-Optimize) sampler.
+
+    Samples posterior related to the inverse problem with Gaussian likelihood and implicit Gaussian prior, and where the forward model is Linear.
+
+    Parameters
+    ------------
+    target : `cuqi.distribution.Posterior`
+        See `cuqi.sampler.LinearRTO`
+
+    x0 : `np.ndarray` 
+        Initial point for the sampler. *Optional*.
+
+    maxit : int
+        Maximum number of iterations of the inner FISTA solver. *Optional*.
+        
+    stepsize : string or float
+        If stepsize is a string and equals either "automatic", then the stepsize is automatically estimated based on the spectral norm.
+        If stepsize is a float, then this stepsize is used.
+
+    abstol : float
+        Absolute tolerance of the inner FISTA solver. *Optional*.
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is `callback(sample, sample_index)`,
+        where `sample` is the current sample and `sample_index` is the index of the sample.
+        An example is shown in demos/demo31_callback.py.
+        
+    """
+    def __init__(self, target, initial_point=None, maxit=100, stepsize = "automatic", abstol=1e-10, adaptive = True, **kwargs):
+
+        if not callable(target.prior.proximal):
+            raise TypeError("Projector needs to be callable")
```

Comment:
Add to target setter after other suggested update.

Reply:
I move this part to the setter of RegularizedLinearRTO's `target`